x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/cluster/routing/allocation/DataTierAllocationDeciderTests.java

```diff
@@ -257,21 +347,37 @@ public void testPreferredTierAvailable() {
             final DesiredNodes desiredNodes = randomBoolean()
                 ? null
                 : createDesiredNodesWithPendingNodes(HOT_DESIRED_NODE, WARM_DESIRED_NODE, COLD_DESIRED_NODE);
+            final var shutdownMetadata = new NodesShutdownMetadata(Collections.emptyMap());
```

Comment:
Can we make this randomly contain some of the desired nodes too to ensure it does not influence the decision? Also sometimes using irrelevantNodesShutdownMetadata would be good.

Reply:
👍🏼 I made this test generate much more broad node shutdown metadata, including the existing irrelevant node shutdown metadata as well as (when desired nodes are present) picking node IDs out of the cluster and desired node external IDs to have as marked for removal to make sure those don't influence the decision.